addons/website/static/src/components/fields/fields.js

```diff
@@ -22,6 +22,25 @@ class PageUrlField extends UrlField {
     setup() {
         super.setup();
         this.serverUrl = `${window.location.origin}/`;
+        this.inputRef = useRef("input");
+
+        // Trigger onchange api on input event to display redirection
+        // parameters as soon as the user types.
+        useEffect(
+            (inputEl) => {
+                if (inputEl) {
+                    const fireChangeEvent = () => {
+                        inputEl.dispatchEvent(new Event("change"));
+                    };
+
+                    inputEl.addEventListener("input", fireChangeEvent);
+                    return () => {
+                        inputEl.removeEventListener("input", fireChangeEvent);
+                    };
+                }
+            },
+            () => [this.inputRef.el],
+        );
```

Comment:
Can't do that with a t-on-click ?

Reply:
The `useEffect` ensures that if the element changes (i.e. re-render), the event listener is still re-applied.

Also, unless I'm mistaken, it is possible to type in a text input without clicking on it first (i.e. tab navigation).

Reply:
> Also, unless I'm mistaken, it is possible to type in a text input without clicking on it first (i.e. tab navigation).

I wrote `t-on-click` but obviously meant `t-on-input`... :grimacing: 

> The useEffect ensures that if the element changes (i.e. re-render), the event listener is still re-applied.

... I sure hope it's the case when adding event handlers with the `t-on-` syntax... otherwise we can just throw away the owl lib? :joy:

Reply:
Oh right, indeed! I really badly misunderstood your first message :sweat_smile: 

Indeed, this makes a lot of sense. I'll make the change

Reply:
In the end: keeping this because t-on-input would fire the event too quickly... to revisist later :shrug: